core/src/main/java/org/opensearch/sql/expression/function/udf/ToNumberFunction.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.expression.function.udf;
+
+import java.util.List;
+import org.apache.calcite.adapter.enumerable.NotNullImplementor;
+import org.apache.calcite.adapter.enumerable.NullPolicy;
+import org.apache.calcite.adapter.enumerable.RexToLixTranslator;
+import org.apache.calcite.linq4j.function.Strict;
+import org.apache.calcite.linq4j.tree.Expression;
+import org.apache.calcite.linq4j.tree.Expressions;
+import org.apache.calcite.rex.RexCall;
+import org.apache.calcite.sql.type.ReturnTypes;
+import org.apache.calcite.sql.type.SqlReturnTypeInference;
+import org.opensearch.sql.calcite.utils.PPLOperandTypes;
+import org.opensearch.sql.expression.function.ImplementorUDF;
+import org.opensearch.sql.expression.function.UDFOperandMetadata;
+
+/**
+ * A custom implementation of number/boolean to string .
+ *
+ * <p>This operator is necessary because tostring has following requirements "binary" Converts a
+ * number to a binary value. "hex" Converts the number to a hexadecimal value. "commas" Formats the
+ * number with commas. If the number includes a decimal, the function rounds the number to nearest
+ * two decimal places. "duration" Converts the value in seconds to the readable time format
+ * HH:MM:SS. if not format parameter provided, then consider value as boolean
+ */
+public class ToNumberFunction extends ImplementorUDF {
+  public ToNumberFunction() {
+    super(
+        new org.opensearch.sql.expression.function.udf.ToNumberFunction.ToNumberImplementor(),
+        NullPolicy.ANY);
+  }
+
+  @Override
+  public SqlReturnTypeInference getReturnTypeInference() {
+    return ReturnTypes.DOUBLE_FORCE_NULLABLE;
+  }
+
+  @Override
+  public UDFOperandMetadata getOperandMetadata() {
+    return PPLOperandTypes.STRING_OR_STRING_INTEGER;
+  }
+
+  public static class ToNumberImplementor implements NotNullImplementor {
+
+    @Override
+    public Expression implement(
+        RexToLixTranslator translator, RexCall call, List<Expression> translatedOperands) {
+      Expression fieldValue = translatedOperands.get(0);
+      int base = 10;
+      if (translatedOperands.size() > 1) {
+        Expression baseExpr = translatedOperands.get(1);
+        return Expressions.call(ToNumberFunction.class, "toNumber", fieldValue, baseExpr);
+      } else {
+        return Expressions.call(ToNumberFunction.class, "toNumber", fieldValue);
+      }
+    }
+  }
+
+  @Strict
+  public static Number toNumber(String numStr) {
+    return toNumber(numStr, 10);
+  }
+
+  @Strict
+  public static Number toNumber(String numStr, int base) {
```

Comment:
simpliy code logic, [not just remove UT](https://github.com/opensearch-project/sql/pull/4605/commits/b5346a6dcfb385f040cc103f8ad480678dc0156a).

* if base=10, convert to int / double, depend on whether numStr contain `.`
* if base!=10,  convert to long, e.g. new BigInteger("FF", base)
* return null value is numStr is malformatted